engine/engine.go

```diff
@@ -358,8 +373,26 @@ func (q *Query) Explain() *ExplainOutputNode {
 	return explainVector(q.exec)
 }
 
-func (q *Query) Profile() {
-	// TODO(bwplotka): Return profile.
+func (q *Query) Analyze() *AnalyzeOutputNode {
+	if observableRoot, ok := q.exec.(model.ObservableVectorOperator); ok {
+		return analyzeVector(observableRoot)
+	}
+	return nil
+}
+
+func analyzeVector(obsv model.ObservableVectorOperator) *AnalyzeOutputNode {
+	telemetry, obsVectors := obsv.Analyze()
+	name, _ := obsv.(model.VectorOperator).Explain()
```

Comment:
This might add some additional overhead, but we can iterate on that.

Reply:
We can return name as well in `Analyze() (string, OperatorTelemetry, []ObservableVectorOperator)` for each operator ?

Reply:
I think we can add the name to the struct and add a method to `OperatorTelemetry` like `Name() string` which would return the operator's name. :thinking: I think that would make the most sense